feat: add support for Yatch YDHS01

afonsofonbraga opened the pull request:
# What is this PR for
<!-- A clear description of what this PR do and the changes made.
If you want you can also add the shortcut link here
- [ ] [Shortcut](http://) -->
This PR contains the implementation of the Yatch YDHS01 sensor. I created it but I couldn't test it right now because I had other important things to do with jetson. 
# How I did it
<!-- Explain a little bit of your implementation -->

# Results, How I tested
This part still needs to be tested, We was not able to get the serial number of the sensor to differentiate both of them. 

# Checklist
- [x] Assign yourself  in the PR
- [ ] Write unit tests (when relevant)
- [x] Run rubocop and rake tests locally
- [ ] If this PR initialize a CMAKE package please [enable styling and linting](https://github.com/tidewise/wetpaint-buildconf/blob/master/README.md#enabling-styling-and-linting-checks-for-a-cmake-package)
- [x] Analyse if this PR modifies the UI, if so:
    - [ ] Tested Tupan's simulation (Charts) :world_map:
    - [ ] Tested Tupan's simulation (Hud) :joystick:
    - [ ] Tested ROV's simulation :joystick:

Commit:
feat: add support for Yatch YDHS01

higor-melo committed:
feat: add a way to differentiate sensors

Since we have multiple of the same sensor
and we're not able to differentiate them
by their BusID once it's static, we need
to differentiate them by using their variable
source which can be changed using the NGT-01.

higor-melo committed:
fix: changed variable names

As stated on the respective PGNs

higor-melo committed:
chore: linting

Just remember to press Ctrl+Shift+I when
working on c++ code

higor-melo committed:
fix: permit usage of the same variable name

higor-melo committed:
fix: altered source to instance

We are able to change the instance id not
the source one, also we need to convert
the actual_humidity to percentage values.

higor-melo committed:
feat: add Dew Point Temperature support

higor-melo committed:
fix: add port connection reliability method

Altered the switch case order to match the
order which the PGNs are received.
Humidity -> Temperature -> Temperature

rayssinha-soares commented on nmea2000.orogen:
what's a 'reliable connection' in these cases?

rayssinha-soares commented on tasks/YatchYDHS01Task.hpp:
remove dead code

doudou commented:
@higor-melo since i did a review in-person, waiting for the revised version to do it here ...

rayssinha-soares commented on tasks/YatchYDHS01Task.cpp:
```diff
@@ -0,0 +1,94 @@
+    YatchYDHS01TaskBase::updateHook();
+
+    Message msg;
+    while (_msg_in.read(msg) == RTT::NewData) {
```
why did you use `while` instead of `if` ? the task is already in loop

higor-melo committed:
chore: refactoring code

- Created functions for the update of
humidity and temperature
- added new id global variable

higor-melo commented:
> @higor-melo since i did a review in-person, waiting for the revised version to do it here ...

Done it. I think i got all the changes requested. Please let me know if i forgot anything.
Also, when i removed the switch case the values started updating correctly, even when changing the order between temperature/humidity.. I made sure to try the code with the temperature PGN first and humidity still updated normally.

higor-melo committed:
chore: remove dead code
